libraries/joomla/database/driver/mysqli.php

```diff
@@ -557,8 +557,7 @@ public function execute()
 
 		if (!is_object($this->connection))
 		{
-			JLog::add(JText::sprintf('JLIB_DATABASE_QUERY_FAILED', $this->errorNum, $this->errorMsg), JLog::ERROR, 'database');
-			throw new RuntimeException($this->errorMsg, $this->errorNum);
+			throw new RuntimeException('Could not connect to MySQL.');
```

Comment:
If someone can write a test case for it to prove its use (disconnected from this PR), then keep it.  But the way I'm reading the code, it seems like this specific if statement is dead code.  Either `$this->connect()` is going to throw an Exception and this if statement will never get triggered or `$this->connect()` going to set an object/resource/whatever-the-driver-uses to `$this->connection` and this if statement is always going to be true.